Add SPEC9 - Governance (#1)
mdhaber/specs: spec-0009/index.md

---
title: "SPEC 9 — Governance"
number: 9
date: 2024-06-04
author:
  - "Sanket Verma <svsanketverma5@gmail.com>"
discussion: https://discuss.scientific-python.org/t/spec-9-governance/1229
endorsed-by:
---

## Description

Open-source project communities are amalgamations of software and the humans who
use and maintain it. We have various systems, like governance models, which

> define certain roles for participating members in the project, and a code of

Author: "participating members in the project" - consider "project members"

Reviewer: @drammock helped in updating the opening paragraph here: https://github.com/scientific-python/specs/pull/323#discussion_r1628575639.

I think the current version looks good. Let me know what you think.

Author: OK, this comment no longer applies.

conduct, which protects and fosters a safe and collaborative environment for
the community.

> Having a robust system in place ensures the long-term sustainability of the

Author: This is a separate paragraph, but it seems connected to the previous paragraph.

Also, IIUC "the project" does not refer to a particular project, but a hypothetical project. In my first read, this did not seem like the only interpretation; I thought I might have missed which project "the" refers to.

Reviewer: I think you're right. I'd incorporate this line in the paragraph above.

Done. Check this: https://github.com/scientific-python/specs/pull/323/commits/d1fece14ca6802dfc3eed95acbccb94f7ebd4a36.

Author: I don't think appearances of "the project" are confusing anymore.

project.

> The OS communities interact and contribute through various mediums, such as

Author: Maybe define the acronym "OS" when you use the term "Open-source" before.

Reviewer: What's the best place to define that in a SPEC document?

Author: https://github.com/scientific-python/specs/pull/323/files#r1646794032

> GitHub, a chat platform, social media, blog posts, etc., and having guidelines

Author: Currently this sentence could be read with the expectation that the "and" will introduce the final element of the list, but it really introduces an independent clause. Also, "etc." isn't strictly necessary since the list was prefaces with "such as". Consider:
```suggestion
GitHub, a chat platform, social media, and blog posts. Having guidelines
```

Reviewer: Thanks for the suggestion here, @mdhaber.

I'd like to keep 'etc.' here as I've left out some of the mediums through which the community interacts with the project. For example, forums like Discourse, mailing lists, and Zoom meetings (community calls, office hours, etc.).

What do you think about this:

The OS communities interact and contribute through various mediums, _like_ distributed version control systems, chat platforms, social media, blog posts, etc. Having clear guidelines for participation helps maintain a continuous stream of contributions.

Author: See, for example:
https://www.grammarbook.com/blog/definitions/all-about-etc/

> And never use etc. at the end of a series that begins with for example, e.g., including, such as, and the like, because these terms make etc. redundant: they already imply that the writer could offer other examples.

This is the point I was making. I think "like" also indicates that the list is incomplete, so "etc." is redundant.

Reviewer: TIL. Thanks!
Done. Check here: https://github.com/scientific-python/specs/pull/323/commits/5b5def515c2d31e15b6432a21d7a3555975ba839

for contributions helps maintain a continuous stream of contributions.

This SPEC outlines the definition of governance, various open-source governance
models, the steps to choose the right governance model, and what needs to be
done post-adoption.

Please note that these are recommendations and not bylaws. If you have
suggestions/thoughts, we're more than happy to consider them.

### Core Project Endorsement

<!--
Briefly discuss what it means for a core project to endorse this SPEC.
-->

### Ecosystem Adoption

<!--
Briefly discuss what it means for a project to adopt this SPEC.
-->

## Implementation

### Defining governance

> Governance are set of rules or customs that defines who gets to do what (and how

Author: ```suggestion
"Governance" is a set of rules or customs that defines who gets to do what (and how
```

Reviewer: We've removed the definition here: https://github.com/scientific-python/specs/pull/323#discussion_r1628601990 and added this in the 'Description' section.

Thanks to @drammock.

they are supposed to do it).

When you define governance for a project, you need to identify a few things:

- What roles does maintainers, core-team, contributors and community members

>   play in the project?

Author: Sometimes the [Oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/) is used (e.g. next bullet), but it is not used here. Consider using it consistently (or never using it).
```suggestion
- What roles does maintainers, core-team, contributors, and community members
  play in the project?
```

Reviewer: Done. 💯

Author: It looks like the Oxford comma is now used consistently throughout.

> - What duties, privileges, and authority are associate with each role?

Author: ```suggestion
- What duties, privileges, and authority are associated with each role?
```
Consider putting "authority" first so that it doesn't *appear* to conflict with "are". (That is, if only "authority" were present, it would be "What authority *is*...". Since "authority" is one of a group of items, "are" is arguably correct... but nobody is going to question it if it were "What authory, duties, and privileges are...")

- How do people get assigned to and removed from roles?

> - How decisions are made and documented within the project?

Author: To follow the same structure as the other bullets:
```suggestion
- How are decisions made and documented within the project?
```

- What are the methods for resolving conflicts?

There are few **pre-defined models** for governance used across open-source
projects. We'll mention them here for reference:

- BDFL
  - BDFL stands for 'Benevolent Dictator for Life.' Under this structure, one
    person (usually the project's initial author) has the final say on all
    major project decisions. Smaller projects are BDFL by default because
    there are only one or two maintainers.
  - BDFL model template → http://oss-watch.ac.uk/resources/benevolentdictatorgovernancemodel
- Meritocracy / Do-ocracy
  - Decisions are made by contributors who have demonstrated merit through
    their contributions to the project. Merit is often gained through

>     contributions of code, documentation, or other significant efforts.

Author: Not a big deal, but "documentation" relies on "contributions of", but "other significant efforts" does not. Maybe:
```suggestion
    code contributions, documentation improvements, or other significant efforts.
```
to avoid it.

  - Meritocracy model template → http://oss-watch.ac.uk/resources/meritocraticgovernancemodel
- Self appointing council / committee / board
  - This model appoints a board or committee to govern various aspects of a
    project. Those groups are often called the steering committee, committer
    council, technical operation committee, board of directors, etc. This
    model might be helpful for projects that need a sponsoring foundation,

>     and establishing electoral mechanisms is challenging.

Author: I don't understand the relationship between this clause tand the rest of the sentence.

Reviewer: This sentence defines various names for the self-appointed board within open-source projects. Some projects use a steering council, some prefer a board, some prefer a committee and so on.

Author: This is what I was referring to:
https://github.com/scientific-python/specs/pull/323/files#r1646795827

- Electoral
  - The electoral governance model in open-source projects is a structured

>     approach in which the community elects the leadership or decision-making

Author: ```suggestion
  - The community elects the leadership or decision-making
```

    body. This model aims to ensure that those who guide the project's
    direction and make crucial decisions are chosen democratically,
    representing the broader community's interests.
- Corporate backed / Single vendor
  - A single company or organization controls the project, often making
    decisions internally. The company might appoint maintainers and decide

>     the project's direction.

Author: Do you mean "The company might ... decide the project's direction" or the "The company might appoint maintainers who decide..."?

Reviewer: I've removed the 'Corporate backed / Single vendor' governance model as per Jarrod's suggestion.
Check https://github.com/scientific-python/specs/pull/323/commits/750a088acb40e843817d94f770259618ae6f8222.
